crates/scroll/state-commitment/src/test.rs

```diff
@@ -243,26 +242,171 @@ proptest! {
+    ) {
+        let factory = create_test_provider_factory();
+        let consistent_view = ConsistentDbView::new(factory.clone(), None);
+
+        let state = state
+            .into_iter()
+            .map(|(address, ((nonce, balance, code), storage))| {
+                let balance = u256_clear_msb(balance);
+                let account_extension = code
+                    .map(|(_, code_hash, code_size)| {
+                        (code_size, b256_clear_first_byte(code_hash)).into()
+                    })
+                    .or_else(|| Some(Default::default()));
+                let account = Account {
+                    balance,
+                    nonce: nonce.into(),
+                    bytecode_hash: code.as_ref().map(|(code_hash, _, _)| *code_hash),
+                    account_extension,
+                };
+                let storage = storage
+                    .into_iter()
+                    .map(|(slot, value)| {
+                        let slot = b256_clear_first_byte(slot);
+                        let hashed_slot = PoseidonKeyHasher::hash_key(slot);
+                        (hashed_slot, value)
+                    })
+                    .collect::<BTreeMap<_, _>>();
+                let hashed_address = PoseidonKeyHasher::hash_key(address);
+                (hashed_address, (account, storage))
+            })
+            .collect::<BTreeMap<_, _>>();
+
+        let provider_rw = factory.provider_rw().unwrap();
+        let mut hashed_storage_cursor = provider_rw.tx_ref().cursor_write::<tables::HashedStorages>().unwrap();
+        let mut hashed_account_cursor = provider_rw.tx_ref().cursor_write::<tables::HashedAccounts>().unwrap();
+
+        for (hashed_address, (account, storage)) in state.clone() {
+            for (hashed_slot, value) in storage {
+                hashed_storage_cursor.upsert(hashed_address, StorageEntry { key: hashed_slot, value }).unwrap();
+            }
+            hashed_account_cursor.upsert(hashed_address, account).unwrap();
+        }
+        provider_rw.commit().unwrap();
+
+        let expected_state_root = crate::test_utils::state_root(state.into_iter().map(
+            |(hashed_address, (account, storage))| {
+                (
+                    b256_reverse_bits(hashed_address),
+                    (
+                        account,
+                        storage.into_iter().map(|(hashed_slot, value)| {
+                            (b256_reverse_bits(hashed_slot), value)
+                        }),
+                    ),
+                )
+            },
+        ));
+
+        let got =
+            ParallelStateRoot::new(consistent_view, Default::default()).incremental_root().unwrap();
+
+        prop_assert_eq!(
+            expected_state_root,
+            got,
+            "expected_root = {:?}, computed_root = {:?}",
+            expected_state_root,
+            got
+        );
+    }
+
+    #[test]
+    fn test_parallel_state_root_memory(
```

Comment:
I tested to check if this fails with the `pack` instead of `pack_bits` and it does. Let me know if you want me to mimic your test with the initial state.